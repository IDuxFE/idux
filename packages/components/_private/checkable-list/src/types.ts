/**
 * @license
 *
 * Use of this source code is governed by an MIT-style license that can be
 * found in the LICENSE file at https://github.com/IDuxFE/idux/blob/main/LICENSE
 */

import type { VirtualScrollToFn } from '@idux/cdk/scroll'
import type { ExtractInnerPropTypes, ExtractPublicPropTypes, MaybeArray, VKey } from '@idux/cdk/utils'
import type { DefineComponent, HTMLAttributes, PropType } from 'vue'

export interface CheckableListData {
  key?: VKey
  label?: string
  disabled?: boolean
  [key: string]: unknown
}

export interface CheckableListScroll {
  height?: string | number
  fullHeight?: boolean
}

export const checkableListProps = {
  dataSource: Array as PropType<CheckableListData[]>,
  checkable: {
    type: Boolean,
    default: true,
  },
  removable: {
    type: Boolean,
    default: false,
  },
  checked: Function as PropType<(item: CheckableListData) => boolean>,
  customAdditional: { type: Function as PropType<CheckableListCustomAdditional>, default: undefined },
  disabled: Function as PropType<(item: CheckableListData) => boolean>,
  getKey: Function as PropType<(item: CheckableListData) => VKey>,
  labelKey: String,
  virtual: {
    type: Boolean,
    default: false,
  },
  scroll: Object as PropType<CheckableListScroll>,
  onCheckChange: [Function, Array] as PropType<MaybeArray<(item: CheckableListData, checked: boolean) => void>>,
  onRemove: [Function, Array] as PropType<MaybeArray<(item: CheckableListData) => void>>,
  onScroll: [Function, Array] as PropType<MaybeArray<(evt: Event) => void>>,
  onScrolledChange: [Function, Array] as PropType<
    MaybeArray<(startIndex: number, endIndex: number, visibleData: unknown[]) => void>
  >,
  onScrolledBottom: [Function, Array] as PropType<MaybeArray<() => void>>,
} as const

export const checkableListItemProps = {
  checked: {
    type: Boolean,
    default: false,
  },
  checkable: {
    type: Boolean,
    default: true,
  },
  removable: {
    type: Boolean,
    default: false,
  },
  disabled: {
    type: Boolean,
    default: false,
  },
  label: String,
  value: {
    type: [String, Number, Symbol] as PropType<VKey>,
    required: true,
  },
  onCheckChange: [Function, Array] as PropType<MaybeArray<(checked: boolean) => void>>,
  onRemove: [Function, Array] as PropType<MaybeArray<() => void>>,
} as const

export interface CheckableListApi {
  scrollTo: VirtualScrollToFn
}

export type CheckableListProps = ExtractInnerPropTypes<typeof checkableListProps>
export type CheckableListItemProps = ExtractInnerPropTypes<typeof checkableListItemProps>
export type CheckableListPublicProps = ExtractPublicPropTypes<typeof checkableListProps>
export type CheckableListComponent = DefineComponent<
  Omit<HTMLAttributes, keyof CheckableListPublicProps> & CheckableListPublicProps,
  CheckableListApi
>
export type CheckableListInstance = InstanceType<DefineComponent<CheckableListProps, CheckableListApi>>

export type CheckableListCustomAdditional = (options: {
  data: CheckableListData
  index: number
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
}) => Record<string, any> | undefined
