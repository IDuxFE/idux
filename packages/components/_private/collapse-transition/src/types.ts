/**
 * @license
 *
 * Use of this source code is governed by an MIT-style license that can be
 * found in the LICENSE file at https://github.com/IDuxFE/idux/blob/main/LICENSE
 */

import type { ExtractInnerPropTypes, ExtractPublicPropTypes } from '@idux/cdk/utils'
import type { DefineComponent, HTMLAttributes } from 'vue'

import { IxPropTypes } from '@idux/cdk/utils'

export type CollapseTransitionMode = 'height' | 'width'

export const collapseTransitionProps = {
  appear: IxPropTypes.bool.def(false),
  name: IxPropTypes.string,
  mode: IxPropTypes.oneOf<CollapseTransitionMode>(['height', 'width']).def('height'),
  onAfterEnter: IxPropTypes.emit(),
  onAfterLeave: IxPropTypes.emit(),
}

export type CollapseTransitionProps = ExtractInnerPropTypes<typeof collapseTransitionProps>
export type CollapseTransitionPublicProps = ExtractPublicPropTypes<typeof collapseTransitionProps>
export type CollapseTransitionComponent = DefineComponent<
  Omit<HTMLAttributes, keyof CollapseTransitionPublicProps> & CollapseTransitionPublicProps
>
export type CollapseTransitionInstance = InstanceType<DefineComponent<CollapseTransitionProps>>
