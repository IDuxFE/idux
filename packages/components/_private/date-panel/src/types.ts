/**
 * @license
 *
 * Use of this source code is governed by an MIT-style license that can be
 * found in the LICENSE file at https://github.com/IDuxFE/idux/blob/main/LICENSE
 */

import type { ExtractInnerPropTypes, ExtractPublicPropTypes, MaybeArray } from '@idux/cdk/utils'
import type { DefineComponent, HTMLAttributes, PropType } from 'vue'

export const datePanelProps = {
  cellTooltip: Function as PropType<(cell: { value: Date; disabled: boolean }) => string | void>,
  disabledDate: Function as PropType<(date: Date) => boolean>,
  type: {
    type: String as PropType<DatePanelType>,
    default: 'date',
  },
  value: [Date, Array] as PropType<Date | (Date | undefined)[]>,
  activeDate: Date,
  visible: {
    type: Boolean,
    default: undefined,
  },
  isSelecting: {
    type: Boolean,
    default: undefined,
  },

  onCellClick: [Function, Array] as PropType<MaybeArray<(date: Date) => void>>,
  onCellMouseenter: [Function, Array] as PropType<MaybeArray<(date: Date) => void>>,
  'onUpdate:activeDate': [Function, Array] as PropType<MaybeArray<(date: Date) => void>>,
}

export type DatePanelProps = ExtractInnerPropTypes<typeof datePanelProps>
export type DatePanelPublicProps = ExtractPublicPropTypes<typeof datePanelProps>
export type DatePanelComponent = DefineComponent<
  Omit<HTMLAttributes, keyof DatePanelPublicProps> & DatePanelPublicProps
>
export type DatePanelInstance = InstanceType<DefineComponent<DatePanelProps>>

export type DatePanelType = 'date' | 'week' | 'month' | 'quarter' | 'year'

// private
export const panelRowProps = {
  rowIndex: {
    type: Number,
    required: true,
  },
}

export const panelCellProps = {
  rowIndex: {
    type: Number,
    required: true,
  },
  cellIndex: {
    type: Number,
    required: true,
  },
}
