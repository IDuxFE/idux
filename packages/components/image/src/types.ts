import type { DefineComponent } from 'vue'

interface ImageOriginalProps {
  src?: string
  width?: string | number
  height?: string | number
  fallback?: string
  preview?: boolean
  alt?: string
  fit?: string
}
interface ImagePreviewOriginalProps {
  previewSrc?: string
}

export type ImageProps = Readonly<ImageOriginalProps>

export type ImagePreviewProps = Readonly<ImagePreviewOriginalProps>

export type ImageStatus = 'loading' | 'loaded' | 'failed'

export type IxImageComponent = InstanceType<DefineComponent<ImageProps>>
