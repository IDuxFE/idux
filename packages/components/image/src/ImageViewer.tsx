/**
 * @license
 *
 * Use of this source code is governed by an MIT-style license that can be
 * found in the LICENSE file at https://github.com/IDuxFE/idux/blob/main/LICENSE
 */

import type { ImageViewerProps } from './types'
import type { ImageViewerConfig } from '@idux/components/config'
import type { ComputedRef } from 'vue'

import { Transition, computed, defineComponent } from 'vue'

import { CdkPortal } from '@idux/cdk/portal'
import { useControlledProp } from '@idux/cdk/utils'
import { useGlobalConfig } from '@idux/components/config'

import ImageViewerContent from './component/ImageViewerContent'
import { imageViewerProps } from './types'

export default defineComponent({
  name: 'IxImageViewer',
  props: imageViewerProps,
  setup(props) {
    const common = useGlobalConfig('common')
    const config = useGlobalConfig('imageViewer')
    const [visible] = useControlledProp(props, 'visible', false)
    const mergedPrefixCls = computed(() => `${common.prefixCls}-image-viewer`)
    const target = useTarget(props, config, mergedPrefixCls)

    return () => (
      <CdkPortal target={target.value} load={visible.value}>
        <Transition name={`${common.prefixCls}-zoom-big-fast`} appear>
          {visible.value && <ImageViewerContent mergedPrefixCls={mergedPrefixCls.value} {...props} />}
        </Transition>
      </CdkPortal>
    )
  },
})

function useTarget(props: ImageViewerProps, config: ImageViewerConfig, mergedPrefixCls: ComputedRef<string>) {
  return computed(() => props.target ?? config.target ?? `${mergedPrefixCls.value}-container`)
}
