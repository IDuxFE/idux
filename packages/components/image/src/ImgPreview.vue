<template>
  <div class="ix-image-preview">
    <div class="ix-image-preview-mask"></div>
    <div class="ix-image-preview-tools">
      <ul class="ix-preview-tools">
        <li class="ix-tools-item ix-rotate-left" @click="rotateEvent(-1)">
          <IxIcon name="rotate-left" />
        </li>
        <li class="ix-tools-item ix-rotate-right" @click="rotateEvent(1)">
          <IxIcon name="rotate-right" />
        </li>
        <li class="ix-tools-item ix-zoom-in" @click="zoomEvent(1)">
          <IxIcon name="zoom-in" />
        </li>
        <li
          class="ix-tools-item ix-zoom-out"
          :class="isZoomOutDisabled ? 'ix-tools-item-disabled' : ''"
          @click="zoomEvent(-1)"
        >
          <IxIcon name="zoom-out" />
        </li>
        <li class="ix-tools-item ix-close" @click="close">
          <IxIcon name="close" />
        </li>
      </ul>
    </div>
    <div class="ix-image-preview-img">
      <img :src="previewSrc" alt="" :style="{ transform: transform }" />
    </div>
  </div>
</template>

<script lang="ts">
import { computed, defineComponent, ref } from 'vue'

import { IxIcon } from '@idux/components/icon'

import { imagePreviewProps } from './types'

const minScale = 0.2
const initScale = 1.0
const scaleStep = 0.1
const initRotate = 0
const rotateStep = 90

export default defineComponent({
  name: 'IxImgPreview',
  components: { IxIcon },
  props: imagePreviewProps,
  emits: {
    close: () => {
      return true
    },
  },
  setup(props, { emit }) {
    const scale = ref(initScale)
    const rotate = ref(initRotate)
    const transform = computed(() => `scale3d(${scale.value}, ${scale.value}, 1) rotate(${rotate.value}deg)`)
    const isZoomOutDisabled = computed(() => scale.value < minScale)
    const close = () => {
      emit('close')
    }
    const rotateEvent = (direction: number) => {
      rotate.value += direction * rotateStep
    }
    const zoomEvent = (direction: number) => {
      if (scale.value > minScale || direction > 0) {
        scale.value += direction * scaleStep
      }
    }

    return {
      close,
      scale,
      rotate,
      rotateEvent,
      zoomEvent,
      transform,
      isZoomOutDisabled,
    }
  },
})
</script>
