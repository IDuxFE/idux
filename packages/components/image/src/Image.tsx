/**
 * @license
 *
 * Use of this source code is governed by an MIT-style license that can be
 * found in the LICENSE file at https://github.com/IDuxFE/idux/blob/main/LICENSE
 */

import type { ImageProps, ImageStatus } from './types'
import type { ImageConfig } from '@idux/components/config'
import type { CSSProperties, ComputedRef, Ref, Slots } from 'vue'

import { computed, defineComponent, normalizeClass, ref, watch } from 'vue'

import { callEmit } from '@idux/cdk/utils'
import { useGlobalConfig } from '@idux/components/config'
import { IxIcon } from '@idux/components/icon'

import ImageViewer from './ImageViewer'
import { imageProps } from './types'

export default defineComponent({
  name: 'IxImage',
  inheritAttrs: false,
  props: imageProps,
  setup(props, { attrs, slots }) {
    const { class: className, style, ...rest } = attrs
    const common = useGlobalConfig('common')
    const mergedPrefixCls = computed(() => `${common.prefixCls}-image`)
    const config = useGlobalConfig('image')
    const preview = usePreview(props, config)
    const { status, setFailed, setLoaded } = useStatus(props)
    const [viewerVisible, setVisible] = useViewerVisible()
    const { outerClasses, overLayerClasses, imageClasses } = useClasses(
      mergedPrefixCls,
      className as string,
      status,
      preview,
    )

    return () => {
      const imageViewerProps = {
        visible: viewerVisible.value,
        'onUpdate:visible': setVisible,
        images: [props.src],
        ...(props.imageViewer ?? {}),
      }

      return (
        <div class={outerClasses.value} style={style as CSSProperties}>
          <div class={overLayerClasses.value}>
            {renderPreviewIcon(props, slots, mergedPrefixCls, status, preview, setVisible)}
            {renderPlaceholder(slots, mergedPrefixCls, status)}
            {renderFallback(slots, mergedPrefixCls, status)}
            <ImageViewer {...imageViewerProps}></ImageViewer>
          </div>
          <img {...rest} src={props.src} class={imageClasses.value} onLoad={setLoaded} onError={setFailed} />
        </div>
      )
    }
  },
})

function renderPreviewIcon(
  props: ImageProps,
  slots: Slots,
  mergedPrefixCls: ComputedRef<string>,
  status: Ref<ImageStatus>,
  preview: ComputedRef<boolean>,
  setVisible: (visible: boolean) => void,
) {
  return (
    status.value === 'loaded' &&
    preview.value && (
      <span class={`${mergedPrefixCls.value}-preview-wrapper`} onClick={() => setVisible(true)}>
        {slots.previewIcon?.() ?? <IxIcon class={`${mergedPrefixCls.value}-preview-icon`} name="zoom-in" />}
      </span>
    )
  )
}

function renderPlaceholder(slots: Slots, mergedPrefixCls: ComputedRef<string>, status: Ref<ImageStatus>) {
  return (
    status.value === 'loading' &&
    (slots.placeholder?.() ?? <IxIcon class={`${mergedPrefixCls.value}-placeholder`} name="loading" />)
  )
}

function renderFallback(slots: Slots, mergedPrefixCls: ComputedRef<string>, status: Ref<ImageStatus>) {
  return (
    status.value === 'failed' &&
    (slots.fallback?.() ?? <IxIcon class={`${mergedPrefixCls.value}-fallback`} name="file-image" />)
  )
}

function useViewerVisible(): [Ref<boolean>, (visible: boolean) => void] {
  const viewerVisible = ref(false)
  const setVisible = (visible: boolean) => {
    viewerVisible.value = visible
  }
  return [viewerVisible, setVisible]
}

function useClasses(
  mergedPrefixCls: ComputedRef<string>,
  className: string,
  status: Ref<ImageStatus>,
  preview: ComputedRef<boolean>,
) {
  const outerClasses = computed(() =>
    normalizeClass([
      mergedPrefixCls.value,
      className,
      `${mergedPrefixCls.value}-${status.value}`,
      { [`${mergedPrefixCls.value}-preview`]: preview.value },
    ]),
  )
  const overLayerClasses = computed(() => normalizeClass(`${mergedPrefixCls.value}-layer`))
  const imageClasses = computed(() =>
    normalizeClass([
      `${mergedPrefixCls.value}-inner`,
      { [`${mergedPrefixCls.value}-inner-hidden`]: status.value !== 'loaded' },
    ]),
  )

  return {
    outerClasses,
    overLayerClasses,
    imageClasses,
  }
}

function usePreview(props: ImageProps, config: ImageConfig) {
  return computed(() => props.preview ?? config.preview)
}

function useStatus(props: ImageProps) {
  const status: Ref<ImageStatus> = ref('loading')
  const setLoaded = (e: Event) => {
    status.value = 'loaded'
    callEmit(props.onLoad, e)
  }
  const setFailed = (e: Event) => {
    status.value = 'failed'
    callEmit(props.onError, e)
  }

  watch(
    () => props.src,
    () => {
      status.value = 'loading'
    },
    { immediate: true },
  )

  return {
    status,
    setLoaded,
    setFailed,
  }
}
