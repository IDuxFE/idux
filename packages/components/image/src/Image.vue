<template>
  <div class="ix-image">
    <img
      v-if="imageStatus !== 'failed'"
      :class="['ix-image-img', preview ? 'ix-image-preview-is' : '']"
      :src="src"
      :style="{ width: imageWidth, height: imageHeight, objectFit: objectFit }"
      :alt="alt"
      @color-error="onError"
      @load="onLoaded"
      @click="onClick"
    />
    <div
      v-else-if="imageStatus === 'failed'"
      class="ix-image-error"
      :style="{
        width: imageWidth,
        height: imageHeight,
        objectFit: objectFit,
        background: `url(${fallback || imageConfig.fallback})`,
      }"
    ></div>
    <template v-if="imageStatus !== 'failed' && preview">
      <img-preview v-if="isShowPreview" :previewSrc="src" @close="onClose"></img-preview>
    </template>
  </div>
</template>
<script lang="ts">
import type { ImageConfig } from '@idux/components/config'
import type { ImageProps, ImageStatus } from './types'

import { defineComponent, computed, ref, watchEffect, watch } from 'vue'
import { convertCssPixel } from '@idux/cdk/utils'
import { useGlobalConfig } from '@idux/components/config'
import ImgPreview from './ImgPreview.vue'
import { imageProps } from './types'

export default defineComponent({
  name: 'IxImage',
  components: { ImgPreview },
  props: imageProps,
  emits: ['statusChange'],
  setup(props, { emit }) {
    const isShowPreview = ref(false)
    const imageConfig = useGlobalConfig('image')
    const imageWidth = computedSize(props, imageConfig, 'width')
    const imageHeight = computedSize(props, imageConfig, 'height')
    const imageStatus = ref<ImageStatus>('loading')
    const onError = () => {
      imageStatus.value = 'failed'
    }
    const onLoaded = () => {
      imageStatus.value = 'loaded'
    }
    const onClick = () => {
      isShowPreview.value = true
    }
    const onClose = () => {
      isShowPreview.value = false
    }
    watchEffect(() => {
      const status = imageStatus.value
      emit('statusChange', status)
    })
    watch(
      () => props.src,
      () => {
        imageStatus.value = 'loading'
      },
    )
    return {
      imageWidth,
      imageHeight,
      onLoaded,
      onError,
      onClick,
      isShowPreview,
      onClose,
      imageStatus,
      imageConfig,
    }
  },
})

const computedSize = (props: ImageProps, imageConfig: ImageConfig, type: 'width' | 'height') => {
  return computed(() => {
    const size = props[type] ?? imageConfig[type]
    return convertCssPixel(size)
  })
}
</script>
