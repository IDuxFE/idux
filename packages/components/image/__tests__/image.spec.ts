import { MountingOptions, flushPromises, mount } from '@vue/test-utils'
import { h } from 'vue'

import { renderWork } from '@tests'

import Image from '../src/Image'
import ImageViewer from '../src/ImageViewer'
import { ImageProps } from '../src/types'

describe('Image', () => {
  const ImageMount = (options?: MountingOptions<Partial<ImageProps>>) => {
    const { props, ...rest } = options ?? {}
    const src = '/icons/logo.svg'

    return mount(Image, { ...({ ...rest, props: { src, ...props } } as MountingOptions<ImageProps>) })
  }

  renderWork<ImageProps>(Image, { props: { src: '/icons/logo.svg' } })

  test('src work', async () => {
    const testSrcA = '/a'
    const testSrcB = '/b'
    const wrapper = ImageMount({ props: { src: testSrcA } })
    await flushPromises()

    expect(wrapper.find('.ix-image-inner').attributes('src')).toBe(testSrcA)

    await wrapper.setProps({ src: testSrcB })

    expect(wrapper.find('.ix-image-inner').attributes('src')).toBe(testSrcB)
  })

  test('preview work', async () => {
    const wrapper = ImageMount()
    await flushPromises()
    await wrapper.find('.ix-image-inner').trigger('load')

    expect(wrapper.find('.ix-image-preview-wrapper').exists()).toBe(true)

    await wrapper.find('.ix-image-preview-wrapper').trigger('click')

    expect(wrapper.findComponent(ImageViewer).props('visible')).toBe(true)

    await wrapper.setProps({ preview: false })

    expect(wrapper.find('.ix-image-preview-wrapper').exists()).toBe(false)
  })

  test('img attrs work', async () => {
    const wrapper = ImageMount({ attrs: { alt: 'testAlt', width: '200' } })
    await flushPromises()

    expect(wrapper.find('.ix-image-inner').attributes('alt')).toBe('testAlt')
    expect(wrapper.find('.ix-image-inner').attributes('width')).toBe('200')
  })

  test('hooks work', async () => {
    const onLoad = vi.fn()
    const onError = vi.fn()
    const wrapper = ImageMount({ props: { onLoad, onError } })
    await flushPromises()
    await wrapper.find('.ix-image-inner').trigger('load')

    expect(onLoad).toBeCalled()

    await wrapper.find('.ix-image-inner').trigger('error')

    expect(onError).toBeCalled()
  })

  test('imageViewerProps work', async () => {
    const imageViewer = {
      visible: true,
      activeIndex: 0,
      images: ['/a'],
      zoom: [1, 2],
      loop: false,
      maskClosable: false,
      target: 'ix-image-container',
      'onUpdate:visible': () => {},
      'onUpdate:activeIndex': () => {},
    }
    const wrapper = ImageMount({ props: { imageViewer } })
    await flushPromises()

    expect(wrapper.findComponent(ImageViewer).props()).toEqual(imageViewer)
  })

  test('slots work', async () => {
    const wrapper = ImageMount({
      slots: {
        previewIcon: h('div', { class: 'slot-previewIcon' }),
        placeholder: h('div', { class: 'slot-placeholder' }),
        fallback: h('div', { class: 'slot-fallback' }),
      },
    })
    await flushPromises()

    expect(wrapper.find('.slot-placeholder').exists()).toBe(true)

    await wrapper.find('.ix-image-inner').trigger('load')

    expect(wrapper.find('.slot-previewIcon').exists()).toBe(true)

    await wrapper.find('.ix-image-inner').trigger('error')

    expect(wrapper.find('.slot-fallback').exists()).toBe(true)
  })
})
