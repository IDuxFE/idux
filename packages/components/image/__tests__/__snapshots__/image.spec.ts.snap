// Vitest Snapshot v1

exports[`Image > render work 1`] = `
"<div class=\\"ix-image ix-image-loading ix-image-preview\\">
  <div class=\\"ix-image-layer\\">
    <!----><i class=\\"ix-icon ix-icon-loading ix-image-placeholder\\" role=\\"img\\" aria-label=\\"loading\\"></i>
    <!---->
    <!---->
  </div><img src=\\"/icons/logo.svg\\" class=\\"ix-image-inner ix-image-inner-hidden\\">
</div>"
`;
